test/smoke/tests.go

```diff
@@ -37,143 +36,125 @@ import (
 				GenerateName: "smoke-test-policy-",
-				Labels:       map[string]string{"foo": "bar"},
 			},
 		}
-		Expect(cl.Create(ctx, &namespace)).NotTo(HaveOccurred())
+		Expect(cl.Create(ctx, &namespace)).To(Succeed())
 
-		By("Creating test SelfSigned Issuer")
-		issuer := cmapi.Issuer{
+		issuer = cmapi.Issuer{
 			ObjectMeta: metav1.ObjectMeta{
 				GenerateName: "smoke-test-policy-",
 				Namespace:    namespace.Name,
 			},
 			Spec: cmapi.IssuerSpec{IssuerConfig: cmapi.IssuerConfig{SelfSigned: &cmapi.SelfSignedIssuer{}}},
 		}
-		Expect(cl.Create(ctx, &issuer)).NotTo(HaveOccurred())
+		Expect(cl.Create(ctx, &issuer)).To(Succeed())
 
-		By("Creating CertificateRequestPolicy for test")
-		policy := policyapi.CertificateRequestPolicy{
+		policy = policyapi.CertificateRequestPolicy{
 			ObjectMeta: metav1.ObjectMeta{
 				GenerateName: "smoke-test-policy-",
 			},
 			Spec: policyapi.CertificateRequestPolicySpec{
-				Allowed: &policyapi.CertificateRequestPolicyAllowed{
-					CommonName: &policyapi.CertificateRequestPolicyAllowedString{Value: ptr.To("*.test.policy")},
-				},
 				Selector: policyapi.CertificateRequestPolicySelector{
 					IssuerRef: &policyapi.CertificateRequestPolicySelectorIssuerRef{
 						Name: ptr.To(issuer.Name),
 					},
 					Namespace: &policyapi.CertificateRequestPolicySelectorNamespace{
-						MatchNames:  []string{"smoke-test-policy-*"},
-						MatchLabels: map[string]string{"foo": "bar"},
+						MatchNames: []string{namespace.Name},
 					},
 				},
 			},
 		}
-		Expect(cl.Create(ctx, &policy)).NotTo(HaveOccurred())
+
+		DeferCleanup(func() {
+			Expect(cl.Delete(ctx, &namespace)).To(Succeed())
+			Expect(cl.Delete(ctx, &policy)).To(Succeed())
+
+			cancel()
+		})
+	})
+
+	JustBeforeEach(func() {
+		Expect(cl.Create(ctx, &policy)).To(Succeed())
 
 		By("Waiting for CertificateRequestPolicy to become ready")
 		Eventually(func() bool {
-			Expect(cl.Get(ctx, client.ObjectKey{Name: policy.Name}, &policy)).NotTo(HaveOccurred())
+			Expect(cl.Get(ctx, client.ObjectKey{Name: policy.Name}, &policy)).To(Succeed())
 			for _, condition := range policy.Status.Conditions {
 				if condition.Type == policyapi.CertificateRequestPolicyConditionReady {
 					return condition.Status == corev1.ConditionTrue && condition.ObservedGeneration == policy.Generation
 				}
 			}
 			return false
 		}).WithTimeout(time.Second * 10).WithPolling(time.Millisecond * 10).Should(BeTrue())
+	})
 
-		By("Binding all authenticated users to the test CertificateRequestPolicy")
-		role := rbacv1.Role{
-			ObjectMeta: metav1.ObjectMeta{
-				GenerateName: "smoke-test-policy-",
-				Namespace:    namespace.Name,
-			},
-			Rules: []rbacv1.PolicyRule{
-				{APIGroups: []string{"policy.cert-manager.io"}, Resources: []string{"certificaterequestpolicies"}, ResourceNames: []string{"smoke-test-policy"}, Verbs: []string{"use"}},
-			},
-		}
-		Expect(cl.Create(ctx, &role)).NotTo(HaveOccurred())
+	Context("with allowed CommonName", func() {
```

Comment:
```suggestion
	Context("CommonName Policy Checks", func() {
```

Because this tests both allowed and denied common names.

Reply:
True, but the policy specifies allowed CNs. The way Ginkgo works, strings are appended into meaningful sentences (BDD) if a test fails. I think we can leave this as is, at least for now. Thanks!